### PR TITLE
Adding driver for FXOS8700CQ accelerometer with userland app

### PR DESCRIPTION
**Pull request by amilich**

### CODE REVIEW

**Comment by amilich** [comment]
There are a couple warnings about how "FXOS8700CQ" should be turned into a snake case name "fxos8700_cq"

File: capsules/src/FXOS8700CQ.rs [hunk 1]
```diff
@@ -0,0 +1,130 @@
+use core::cell::Cell;
+use kernel::{AppId, Callback, Driver};
+use kernel::common::take_cell::TakeCell;
+use kernel::hil::i2c::{I2CDevice, I2CClient, Error};
+
+pub static mut BUF: [u8; 6] = [0; 6];
+
+const DEFAULT_SCALE: u8 = 0x0;
```
Comment (alevy) [inline capsules/src/FXOS8700CQ.rs]: Doesn't seem to be used anywhere

File: capsules/src/FXOS8700CQ.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
+use core::cell::Cell;
```
Comment (alevy) [inline capsules/src/FXOS8700CQ.rs]: Add a descriptive comment at the top: what is this capsule for? what does it do? How do I initialize and use it? (use `//!` style comments)

File: capsules/src/FXOS8700CQ.rs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    WhoAmI = 0x0D,
+    CtrlReg1 = 0x2A,
+}
+
+#[derive(Clone,Copy,PartialEq)]
+enum State {
+    /// Sensor is disabled (but on)
+    Disabled,
+
+    /// Reading that sensor is present
```
Comment (alevy) [inline capsules/src/FXOS8700CQ.rs]: I don't understand this comment

File: capsules/src/FXOS8700CQ.rs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+    /// Reading accelerometer data
+    ReadingAcceleration,
+}
+
+pub struct FXOS8700CQ<'a> {
+    i2c: &'a I2CDevice,
+    state: Cell<State>,
+    buffer: TakeCell<&'static mut [u8]>,
+    callback: Cell<Option<Callback>>,
```
Comment (alevy) [inline capsules/src/FXOS8700CQ.rs]: Hrm... ok, this _should_ be virtualized, but none of the other sensors currently have virtualization, so maybe we should just do them altogether.

File: userland/libtock/FXOS8700CQ.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  int x;
+  int y;
+  int z;
+  bool fired;
+};
+
+// internal callback for faking synchronous reads
+static void FXOS8700CQ_cb(int x, int y, int z, void* ud) {
+  struct fx0_data* result = (struct fx0_data*) ud;
+  printf("callback x= %i y = %i z = %i\n", x, y, z);
```
Comment (alevy) [inline userland/libtock/FXOS8700CQ.c]: Looks like this is leftover from debuggin. please remove

File: userland/libtock/FXOS8700CQ.c [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  err = FXOS8700CQ_start_accel_reading();
+  if (err < 0) {
+    return err;
+  }
+
+  yield_for(&result.fired);
+
+  // TODO add sqrt
```
Comment (alevy) [inline userland/libtock/FXOS8700CQ.c]: Just add a short explanation for what this is blocking on. E.g., "add sqrt once software floating-point is supported in userspace. Currently would crash."

File: userland/libtock/FXOS8700CQ.h [hunk 7]
```diff
@@ -0,0 +1,21 @@
+#ifndef _FXOS8700CQ_H
```
Comment (alevy) [inline userland/libtock/FXOS8700CQ.h]: In general, we're moving to the syntax:

`#pragma once` instead of the `ifndef` style. See `adc.h` for an example.

**Pull request merged**